### PR TITLE
feat(windows): add support for New Architecture

### PR DESCRIPTION
**Pull request by tido64**
### Description

Run `install-windows-test-app --use-fabric` to enable New Architecture on 0.74 or higher. Note that this feature is experimental and incomplete.

It's still missing a menu bar, but I'm saving that for another PR as this one is getting big enough already.

### Platforms affected

- [ ] Android
- [ ] iOS
- [ ] macOS
- [x] Windows

### Test plan

```sh
npm run set-react-version canary-windows
yarn
cd example
yarn install-windows-test-app --use-fabric
yarn windows

# In a separate terminal
yarn start
```

### Screenshots

image

### CODE REVIEW

File: windows/Shared/JSValueWriterHelper.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include <map>
+#include <string>
+#include <vector>
+
+#include <winrt/Microsoft.ReactNative.h>
+#include <winrt/base.h>
+
+namespace ReactApp
+{
+    void JSValueWriterWriteValue(winrt::Microsoft::ReactNative::IJSValueWriter const &writer,
```
Comment (acoates-ms) [inline windows/Shared/JSValueWriterHelper.h:13]: This looks like its basically the same as:

#include <JSValueWriter.h>
winrt::Microsoft::ReactNative::WriteValue(writer, value);

File: windows/UWP/MainPage.cpp [hunk 2]
```diff
@@ -128,10 +97,9 @@ namespace
             [initialProps = component.initialProperties](IJSValueWriter const &writer) {
                 if (initialProps.has_value()) {
```
Comment (acoates-ms) [inline windows/UWP/MainPage.cpp:98]: Similarly, I think `winrt::Microsoft::ReactNative::WriteValue(writer, initialProps)` would do all of this?

File: windows/Win32/Images/LockScreenLogo.scale-200.png
Comment (acoates-ms) [inline windows/Win32/Images/LockScreenLogo.scale-200.png:1]: No cool RNTA icon? -- Surely we can throw a simple prompt at an AI to generate something more interesting?

File: windows/Win32/Main.cpp [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    winrt::ReactViewOptions MakeReactViewOptions(ReactTestApp::Component const &component)
+    {
+        winrt::ReactViewOptions viewOptions;
+        viewOptions.ComponentName(winrt::to_hstring(component.appKey));
+
+        auto initialProps = component.initialProperties.value_or(std::map<std::string, std::any>{});
+        initialProps["concurrentRoot"] = true;
+        viewOptions.InitialProps(
+            [initialProps = std::move(initialProps)](winrt::IJSValueWriter const &writer) {
+                writer.WriteObjectBegin();
```
Comment (acoates-ms) [inline windows/Win32/Main.cpp:67]: winrt::Microsoft::ReactNative::WriteValue(writer, initialProps);

**Review by acoates-ms — APPROVED** [review]

File: windows/Win32/Images/LockScreenLogo.scale-200.png
Comment (tido64) [reply windows/Win32/Images/LockScreenLogo.scale-200.png:1]: I'm going to assign that task to you 😉

Comment (kelset) [reply windows/Win32/Images/LockScreenLogo.scale-200.png:1]: on it 🤓🤣

**Pull request merged**